dynamicio/base.py

```diff
@@ -0,0 +1,126 @@
+# pylint: disable=no-name-in-module disable=invalid-name
+
+"""BaseResource class for creating various resources types."""
+from abc import ABC, abstractmethod
+from typing import Optional, Type, TypeVar
+
+import pandas as pd
+import pandera as pa
+from pandera import SchemaModel
+from pydantic import BaseModel
+
+SchemaType = TypeVar("SchemaType", bound=pa.SchemaModel)  # TODO: utilise this
+
+
+class BaseResource(BaseModel, ABC):
+    """BaseClass for resource classes."""
+
+    pa_schema: Optional[pa.SchemaModel]
+    validate_default: bool = True
+    log_metrics_default: bool = True
+    allow_no_schema: bool = False
+
+    def read(
+        self,
+        validate: Optional[bool] = None,
+        log_metrics: Optional[bool] = None,
+        pa_schema: Optional[Type[SchemaModel]] = None,
```

Comment:
Should we hardcode the usage of specifically `pandera` here? To ease the migration from `v4` to `v5`, what if this could also accept the current yaml schemas that we have? Is there value in that?

Reply:
We could have a yaml_schema option here tha converts the yaml schema to pandera, or we could have some code outside of this that either converts yaml schema to pandera python code and/or a function that loads a given schema and converts it.

Reply:
Either one seems like a decent thing to do to help the migration process 🤔